### PR TITLE
Product Page: Add test for disabled products

### PR DESCRIPTION
**Pull request by batbattur**
## Summary
1. Added a new `msw` handler for disabled products.
2. Added a test for the disabled product page that checks the following:
    - `Not for Sale`, description, and warranty texts are displayed.
    - `Add to cart`, `Notify me`, shipping restriction and stock details texts are **not** displayed.
    - Noindexed
    
Got the test specs from  https://github.com/iFixit/react-commerce/issues/1153.

## QA notes
Passing ci is enough.

qa_req 0
closes: #1163

### CODE REVIEW

**Comment by batbattur** [comment]
deploy_block 🚚 on https://github.com/iFixit/react-commerce/pull/1216 going out first

File: frontend/tests/playwright/product/disabled_product.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      // Assert the following to be not visible
+      await expect(page.getByText('Add to Cart')).not.toBeVisible();
+      await expect(page.getByText(/Only \d left/i)).not.toBeVisible();
+      await expect(page.getByText('Notify me')).not.toBeVisible();
+      await expect(
+         page.getByText('Shipping restrictions apply')
+      ).not.toBeVisible();
+
+      // Assert that we noindex disabled product pages
+      const metaRobots = page.locator('meta[name="robots"]');
```
Comment (ardelato) [inline frontend/tests/playwright/product/disabled_product.spec.ts:22]: oooh nice!

File: frontend/tests/playwright/product/disabled_product.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { test, expect } from '@playwright/test';
+
+test.describe('Disabled Product Test', () => {
+   test('Not for Sale text renders and noindexed', async ({ page }) => {
+      await page.goto('products/iphone-6s-plus-replacement-battery-disabled');
+
+      await expect(page.getByText('Not for Sale')).toBeVisible();
+      await expect(page.getByText('Description')).toBeVisible();
```
Comment (ardelato) [inline frontend/tests/playwright/product/disabled_product.spec.ts:8]: In the original issue, https://github.com/iFixit/react-commerce/issues/1153, it says `no variant details` doesn't this mean that this should `not.toBeVisible()`?

Comment (ardelato) [reply frontend/tests/playwright/product/disabled_product.spec.ts:8]: Ah, scratch that. I see that there is some elaboration later in the issue `We will keep the images, description, warranty, video, etc.`

Comment (ardelato) [reply frontend/tests/playwright/product/disabled_product.spec.ts:8]: We should confirm that `assembly contents` and `kit contents` are not visible then.

**Review by ardelato — APPROVED** [review]
CR 👍 but deploy_block 🌧️ we should add some more assertions to validate

- [ ] No Variant Selector
- [ ] No International Buy Box
- [ ] No Stock Details
- [ ] No Assembly Contents
- [ ] No Kit Contents

**Comment by batbattur** [comment]
> * [ ]  No Variant Selector
> * [ ]  No International Buy Box

Added assertions for the above in https://github.com/iFixit/react-commerce/pull/1230/commits/063de006771fdf14902e4b6cfb512ed272a41204. 

> * [ ]  No Stock Details

I tried to check the stock number with the following assertion 😅 (wasn't sure how else to do this since we display stock quantity only when the stock is low). https://github.com/iFixit/react-commerce/blob/cb6ca5aad0494eed3a74522db689fac9303e100a/frontend/tests/playwright/product/disabled_product.spec.ts#L15 

> * [ ]  No Assembly Contents
> * [ ]  No Kit Contents

It looks like the `Kit Contents` section is displayed for a disabled product, so I assume it's the same for the `Assembly Contents` (disabled product on cominor: https://www.cominor.com/products/iphone-5c-replacement-battery).


un_deploy_block 🚛

**Review by mlahargou — APPROVED** [review]
CR 📱

**Comment by mlahargou** [comment]
deploy_block 📱 on the parent PR going out first.

**Comment by batbattur** [comment]
un_deploy_block 🚛 the parent pull got merged

**Pull request merged**